testdata/p4_16_samples/parser-unroll-test1.p4

```diff
@@ -76,7 +76,9 @@ parser MyParser(packet_in packet,
 
     @name (".parse_srcRouting") state parse_srcRouting {
         packet.extract(hdr.srcRoutes[index]);
-        index = index + 1;
+        // Check arbitrary size int handling in interpreter
+        index = (int<32>)((int)index + 1);
```

Comment:
It seems like this should be an error -- the `int` type is only supposed to apply to constant values (or at least compile-time known constant values).  By the time things get to the backend, everything should have a size, and if you allow something like this to get to the backend, it probably won't be able to handle it.

Reply:
The check in question here happens quite early in the frontend, so I would argue that this legal construct should be accepted by the frontend (and definitely not end in a `BUG`). Then it is up to the backend (or rather specific midend passes used) to assign specific width to `int` or to reject the code (or to allow `int` only for constants or operations where the max width can be deduced).
This is BTW what happens if the same construct is used inside a control block (not a parser) - the frontend accepts it (as the symbolic interpreter is used only to determine parser unrolling) and it is up to the backend to handle it.

Just to be clear - explicitly defining `int` variable or field is forbidden (both in compiler and in specification).

Reply:
Well, it definitely should not crash, and if it should be an error, it should be checked somewhere else, not in ExpressionEvaluator.